zips/draft-str4d-account-metadata-keys.md

```diff
@@ -0,0 +1,198 @@
+        - $\ldots / \ldots$ - (Reserved for future updates to this ZIP)
+        - $\ldots / \texttt{0x7FFFFFFFF}' \# \texttt{PrivateSubject}$ - Private-use External Metadata Key
+      - $\ldots / \ldots$ - (Reserved for future updates to this ZIP)
+    - $\ldots / \ldots$ - (Reserved for future updates to this ZIP)
+
+Non-leaf keys in the key tree MUST NOT be used directly to encrypt metadata.
+Encryption keys are leaves in this key tree. The sole exception to this is
+private-use keys (for which part of their key derivation is outside the scope of
+this specification): encryption keys MAY be derived from the private-use key
+leaves.
+
+### Account Metadata Key
+
+The Account Metadata Key is the root of a subtree that corresponds to a ZIP 32
+account represented elsewhere in the overall tree. It is derived from the seed
+$\mathsf{S}$ as:
+
+$\mathsf{AccountMetadataKey} = \mathsf{CKDreg}(\mathsf{CKDreg}(\mathsf{RegKD}(\texttt{“MetadataKeys”}, \mathsf{S}, \mathsf{ZipNumber}), \mathsf{coinType}, [\,]), \mathsf{account}, [\,])$
+
+or, in path notation:
+
+```
+m_metadata / TBD' / coin_type' / account'
+```
+
+### Inherent metadata keys
+
+The Account-level Inherent Metadata Key's subtree contains keys used for
+metadata associated with the Account Metadata Key's corresponding account. The
+key is derived as:
+
+$\mathsf{CKDreg}(\mathsf{AccountMetadataKey}, 0, [\,])$
+
+### External metadata keys
+
+The Account-level External Metadata Key's subtree contains keys used for
+metadata associated with imported UFVKs. Unlike the inherent metadata keys which
+can leverage the inherent domain separation provided by the account index, here
+domain separation between metadata keys is provided by the UFVKs themselves.
+
+As UFVKs may in general change over time (due to the inclusion of new
+higher-preference FVK items, or removal of older deprecated FVK items), there is
+no guarantee that the exact same set of FVK items will be present at both backup
+creation time and recovery time. Instead, the most-preferred FVK item within a
+UFVK is used as the domain separator, and the Imported UFVK Metadata Key is
+derived as:
+
+$\mathsf{CKDreg}(\mathsf{CKDreg}(\mathsf{AccountMetadataKey}, 1, [\,]), 0, \langle FVKTypedItem \rangle)$
+
+where $\langle FVKTypedItem \rangle$ is the encoding of the most preferred FVK
+item within the ZIP 316 raw encoding of a UFVK. [^zip-0316]
+
+Usage of the Imported UFVK Metadata Key trees SHOULD follow ZIP 316 preference
+order: [^zip-0316]
+
+- For encryption-like usage, the key tree corresponding to the most preferred
+  FVK item within a UFVK SHOULD be used.
+- For decryption-like usage, each key tree SHOULD be tried in preference order
+  until metadata can be recovered. If metadata is recovered via an FVK item that
+  is not the most preferred, wallets SHOULD update their metadata backups by
+  re-encrypting the metadata using the key tree corresponding to most preferred
+  FVK item.
```

Comment:
Clarify how "most preferred" applies to unrecognised items (either here or in ZIP 316). The desired behaviour is what I believe `zcash_address::unified` does, i.e. sort all unrecognised items after recognised items, but in reverse typecode order.